pkg/controller/knativeeventing/knativeeventing_controller.go

```diff
@@ -161,6 +163,31 @@ func (r *ReconcileKnativeEventing) updateStatus(instance *eventingv1alpha1.Knati
 	return nil
 }
 
+// check in order if we update
+func (r *ReconcileKnativeEventing) updateServiceMeshMemberRole(api client.Client) error {
+	smmr := &maistrav1.ServiceMeshMemberRoll{}
+	if err := api.Get(context.TODO(), client.ObjectKey{Namespace: "knative-serving-ingress", Name: "default"}, smmr); err != nil {
```

Comment:
I don't love the hard-coded namespace here, but because we intend to eventually deprecate this operator soon in favor of [this soon-to-be-renamed openshift-specific operator](https://github.com/openshift-knative/knative-serving-openshift/blob/master/pkg/controller/knativeserving/servicemesh/servicemesh.go#L174-L207), I'm not too worried about it now. It's a good heads-up that we'll need to refactor that code to account for the `knative-eventing` ns as well.

Reply:
I fully agree with you here